DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -218,58 +186,47 @@ CentreAnswersData centreAnswersData
                 myAccountDetailsData.ProfileImage,
                 myAccountDetailsData.ProfessionalRegistrationNumber,
                 myAccountDetailsData.HasBeenPromptedForPrn,
-                myAccountDetailsData.JobGroupId!.Value,
+                myAccountDetailsData.JobGroupId,
                 detailsLastChecked,
-                userId
+                myAccountDetailsData.UserId
             );
 
-            userDataService.SetCentreEmail(
-                userId,
-                centreAnswersData.CentreId,
-                centreAnswersData.CentreEmail
-            );
+            if (centreId != null)

```

Comment:
Surely this should be if centreEmail not null?

Reply:
No, we need to be able to clear centre email

Reply:
Aha, gotcha. Guessing you've left the centreId nullable in preparation for the centre-independent EditDetails journey.
I think we want to be using a new method for that anyway, so I think it'd be better to have centreId not-nullable.

Reply:
For now it's fine if we don't save anything when User.GetCentreId() returns null.